extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java

```diff
@@ -82,15 +84,27 @@ void validateBeanDeployment(
             }
         }
 
-        for (InjectionPointInfo injectionPoint : validationPhase.getContext().get(BuildExtension.Key.INJECTION_POINTS)) {
+        for (InjectionPointInfo injectionPoint : validationPhase.getContext()
+                .get(BuildExtension.Key.INJECTION_POINTS)) {
             if (injectionPoint.getRequiredType().name().equals(NAME_EMITTER)) {
                 AnnotationInstance stream = injectionPoint.getRequiredQualifier(NAME_STREAM);
                 if (stream != null) {
                     // Stream.value() is mandatory
                     String name = stream.value().asString();
+                    AnnotationInstance annotation = injectionPoint.getTarget().asField()
```

Comment:
We should also support constructor and initializer/setter injection...

Reply:
By the way, the `OnOverflow`'s javadoc is using wrong annotation name in the example: `OnBackPressure`.

Reply:
> We should also support constructor and initializer/setter injection...

what do you mean?

Reply:
An injection point does not have to be a field ;-). You can also use a bean constructor or initializer methods (e.g. a setter) -> in that case, `InjectionPointInfo.getTarget()` returns a `MethodInfo`; `InjectionPointInfo.getPosition()` can be used to obtain the parameter annotations. `io.quarkus.arc.processor.Annotations.getParameterAnnotations(BeanDeployment, MethodInfo, int)` is a convenient util method to get the annotations including those added by means of `AnnotationsTransformer`.

Reply:
Oh I see...

Reply:
Ok should be fixed now.

Reply:
so it was NOT fixed.

Reply:
now it's fixed, and I've added a test to ensure it's working.